pkg/cache/repository.go

```diff
@@ -319,6 +319,7 @@ func (r *cachedRepository) DeletePackage(ctx context.Context, old repository.Pac
 }
 
 func (r *cachedRepository) Close() error {
+	r.pollOnce(context.TODO())
```

Comment:
Ye my only concern is the poll is an expensive operation and we need to reduce the number of redundant polls imho.
The mutex is held for the routine and this may cause other issues no?
I see there is already a TODO comment here regarding the main problem.
https://github.com/nephio-project/porch/blob/main/pkg/cache/repository.go#L399

Reply:
Also consider that there may (and should) be several identical pods of Porch running simultaneously in a ReplicaSet. The mutex doesn't save us from multiple processes updating their independent caches at the same time.

Reply:
Also, if you receive a "please apply your changes to the latest revision" kind of error in metadatastore, that just means optimistic concurrency kicked in inside the kube-apiserver. metadatastore should handle it, and re-read the PkgRev object from kube-apiserver and retry the removal of the finalizer.

Reply:
Thanks for the pointer, @nagygergo - I think I've found a good place to implement that, in crdMetadataStore.Delete(). Wrapping the get/update section there in a RetryOnConflict seems to resolve the orphaned PackageRevs, so I'll remove the polling from here. @efiacor, @tliron, I think that resolves your concerns as well?